src/HealthGPS.Datastore/datamanager.cpp

```diff
@@ -1,86 +1,60 @@
 
+#include <fstream>
 #include <utility>
 
 namespace {
-using namespace jsoncons;
-
-json resolve_uri(const jsoncons::uri &uri, const std::filesystem::path &schema_directory) {
-    constexpr const char *url_prefix =
-        "https://raw.githubusercontent.com/imperialCHEPI/healthgps/main/schemas/v1/";
-
-    const auto &uri_str = uri.string();
-    if (!uri_str.starts_with(url_prefix)) {
-        throw std::runtime_error(fmt::format("Unable to load URL: {}", uri_str));
-    }
-
-    // Strip URL prefix and load file from local filesystem
-    const auto filename = std::filesystem::path{uri.path()}.filename();
-    const auto schema_path = schema_directory / filename;
-    auto ifs = std::ifstream{schema_path};
-    if (!ifs) {
-        throw std::runtime_error("Failed to read schema file");
-    }
-
-    return json::parse(ifs);
-}
-
-void validate_index(const std::filesystem::path &schema_directory, std::ifstream &ifs) {
-    // **YUCK**: We have to read in the data with jsoncons here rather than reusing
-    // the nlohmann-json representation :-(
-    const auto index = json::parse(ifs);
-
-    // Load schema
-    auto ifs_schema = std::ifstream{schema_directory / "data_index.json"};
-    if (!ifs_schema) {
-        throw std::runtime_error("Failed to load schema");
-    }
-
-    const auto resolver = [&schema_directory](const auto &uri) {
-        return resolve_uri(uri, schema_directory);
-    };
-    const auto schema = jsonschema::make_json_schema(json::parse(ifs_schema), resolver);
-
-    // Perform validation
-    schema.validate(index);
-}
-} // namespace
-
-namespace hgps::data {
-DataManager::DataManager(std::filesystem::path root_directory, VerboseMode verbosity)
-    : root_{std::move(root_directory)}, verbosity_{verbosity} {
-    auto full_filename = root_ / "index.json";
+nlohmann::json read_input_files_from_directory(const std::filesystem::path &root_directory) {
+    auto full_filename = root_directory / "index.json";
     auto ifs = std::ifstream{full_filename};
     if (!ifs) {
         throw std::invalid_argument(
             fmt::format("File-based store, index file: '{}' not found.", full_filename.string()));
     }
 
-    index_ = nlohmann::json::parse(ifs);
+    auto index = nlohmann::json::parse(ifs);
 
-    if (!index_.contains("version")) {
+    if (!index.contains("version")) {
         throw std::runtime_error("File-based store, invalid definition missing schema version");
     }
 
-    auto version = index_["version"].get<int>();
+    auto version = index["version"].get<int>();
     if (version != 2) {
         throw std::runtime_error(fmt::format(
             "File-based store, index schema version: {} mismatch, supported: 2", version));
     }
 
     // Validate against schema
     ifs.seekg(0);
-    const auto schema_directory = get_program_directory() / "schemas" / "v1";
-    validate_index(schema_directory, ifs);
+    const auto schema_directory = hgps::get_program_directory() / "schemas" / "v1";
```

Comment:
This is odd. In the previous line you raise an error if the schema version is not 2, but here you are loading schema v1. Confusing, to say the least. Is this an error?

Reply:
It's not. This data file has had a `version` property for ages now (and I think some of the other input files have one too) but the problem is that it doesn't seem like they were particularly careful with backwards compatibility, so the field is basically meaningless anyway.

I was going to remove this property because it's pointless now that we're using schemas. Shall I open an issue for it?

Reply:
Indeed!

Reply:
https://github.com/imperialCHEPI/healthgps/issues/399